src/api/metadata/index.ts

```diff
@@ -1,24 +1,38 @@
 import log from 'loglevel'
 import { Context } from '../../utils/context'
 import { getSerializedCID, loadIpfsFromCid } from '../../utils/appData'
-import { AppDataDoc } from './types'
+import { pinJSONToIPFS } from '../../utils/ipfs'
+import { AppDataDoc, MetadataDoc } from './types'
 import { CowError } from '../../utils/common'
 
+const DEFAULT_APP_CODE = 'CowSwap'
+
 export class MetadataApi {
   context: Context
 
   constructor(context: Context) {
     this.context = context
   }
 
+  generateAppDataDoc(metadata: MetadataDoc = {}): AppDataDoc {
+    return {
+      version: '0.1.0',
+      appCode: DEFAULT_APP_CODE,
```

Comment:
I think it makes sense to have the appCode customizable as an optional parameter